### PR TITLE
Tests: add script for running tests in docker

### PR DESCRIPTION
**Pull request by brettdh**
Just a little something I found useful while working on PyInstaller
from a mac; this lets me easily run the tests on Linux.

### CODE REVIEW

File: tests/scripts/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ADD requirements.txt tests/requirements-*.txt /src/
+RUN pip install \
+    -r /src/requirements.txt \
+    -r /src/requirements-tools.txt \
+    -r /src/requirements-libraries.txt
+
+ADD *.py setup.cfg README.rst MANIFEST.in .pylintrc /src/
+ADD PyInstaller /src/PyInstaller
+ADD bootloader /src/bootloader
+ADD doc /src/doc
```
Comment (htgoebel) [inline tests/scripts/Dockerfile]: Is this used at all in the container? I assume this can be removed.

File: tests/scripts/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+RUN rm -rf /src/PyInstaller/bootloader/*
+
+WORKDIR /src/bootloader
+RUN python waf distclean all
+
+WORKDIR /src
+RUN python setup.py install
+
+ADD tests /src/tests
```
Comment (htgoebel) [inline tests/scripts/Dockerfile]: Should go to the other "ADD" commands.

File: tests/scripts/Dockerfile (hunk 1 above)
Comment (brettdh) [reply tests/scripts/Dockerfile]: I only added things which were required for one of these steps:

- `python setup.py install`
- `python waf distclean all` (in `bootloader/`)
- `py.test tests/functional`

I forget which step needed `doc/` to exist, but one of them did.

File: tests/scripts/Dockerfile (hunk 2 above)
Comment (brettdh) [reply tests/scripts/Dockerfile]: This was intentionally ordered to avoid e.g. redoing `setup.py install` if only test files have changed.

Comment (htgoebel) [reply tests/scripts/Dockerfile]: IC. I always forgot docker is not declarative. /(I'm using guix for quite some while now which is declarative.)

**Pull request merged**